pkg/kafka/consumer/config.go

```diff
@@ -52,24 +51,22 @@ type Configuration struct {
 
 // NewConsumer creates a new kafka consumer
 func (c *Configuration) NewConsumer(logger *zap.Logger) (Consumer, error) {
-	saramaConfig := cluster.NewConfig()
-	saramaConfig.Group.Mode = cluster.ConsumerModePartitions
+	saramaConfig := sarama.NewConfig()
 	saramaConfig.ClientID = c.ClientID
 	saramaConfig.RackID = c.RackID
 	if len(c.ProtocolVersion) > 0 {
 		ver, err := sarama.ParseKafkaVersion(c.ProtocolVersion)
 		if err != nil {
 			return nil, err
 		}
-		saramaConfig.Config.Version = ver
+		saramaConfig.Version = ver
 	}
-	if err := c.AuthenticationConfig.SetConfiguration(&saramaConfig.Config, logger); err != nil {
+	if err := c.AuthenticationConfig.SetConfiguration(saramaConfig, logger); err != nil {
 		return nil, err
 	}
-	// cluster.NewConfig() uses sarama.NewConfig() to create the config.
-	// However the Jaeger OTEL module pulls in newer samara version (from OTEL collector)
-	// that does not set saramaConfig.Consumer.Offsets.CommitInterval to its default value 1s.
-	// then the samara-cluster fails if the default interval is not 1s.
-	saramaConfig.Consumer.Offsets.CommitInterval = time.Second
-	return cluster.NewConsumer(c.Brokers, c.GroupID, []string{c.Topic}, saramaConfig)
+	client, err := sarama.NewConsumerGroup(c.Brokers, c.GroupID, saramaConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error creating consumer group client: %w", err)
+	}
+	return client, err
```

Comment:
I prefer to be explicit since we know `err` is `nil`.

```suggestion
	return client, nil
```